api/apiclient_test.go

```diff
@@ -84,25 +86,29 @@ func (s *apiclientSuite) TestDialAPIMultiple(c *gc.C) {
 }
 
 func (s *apiclientSuite) TestDialAPIMultipleError(c *gc.C) {
-	listener, err := net.Listen("tcp", "127.0.0.1:0")
-	c.Assert(err, jc.ErrorIsNil)
-	defer listener.Close()
+	var addrs []string
+
 	// count holds the number of times we've accepted a connection.
 	var count int32
-	go func() {
-		for {
-			client, err := listener.Accept()
-			if err != nil {
-				return
+	for i := 0; i < 3; i++ {
```

Comment:
I'm curious about this. How did this test get three connection attempts prior to this PR? Because there were three addresses set up in `s.APIInfo(c)`?